service-tests/github.js

```diff
@@ -9,13 +9,50 @@ const {
   isFormattedDate,
   isVPlusDottedVersionAtLeastOne
 } = require('./helpers/validators');
+const colorscheme = require('../lib/colorscheme.json');
 
 const t = new ServiceTester({ id: 'github', title: 'Github' });
 module.exports = t;
-
-t.create('License')
-  .get('/license/badges/shields.json')
-  .expectJSONTypes(Joi.object().keys({ name: 'license', value: Joi.string() }));
+const colorsB = Object.assign({}, ...Object.keys(colorscheme).map(color => ({ [color]: colorscheme[color].colorB })));
+
+t.create('Public domain license')
+  .get('/license/badges/shields.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'CC0-1.0', colorB: '#7cd958' });
+
+t.create('Copyleft license')
+  .get('/license/ansible/ansible.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'GPL-3.0', colorB: colorsB.orange });
+
+t.create('Permissive license')
+  .get('/license/atom/atom.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'MIT', colorB: colorsB.green });
+
+t.create('License for repo without a license')
+  .get('/license/badges/badger.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'missing', colorB: colorsB.red });
+
+t.create('License for repo with an unrecognized license')
+  .get('/license/philokev/sopel-noblerealms.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'unknown', colorB: colorsB.lightgrey });
+
+t.create('License with SPDX id not appearing in configuration')
+  .get('/license/user1/project-with-EFL-license.json?style=_shields_test')
+  .intercept(nock => nock('https://api.github.com')
+    .get(/\/repos\/user1\/project-with-EFL-license/)
```

Comment:
I think you could make this a string instead of a regex: `'/repos/user1/project-with-EFL-license'`

Reply:
Query params can be added to request url (`client_id` and `client_secret`) and nock matches mocks by path and query params. But there is `.query(true)`, which set nock to ignore query params.